lib/matplotlib/colorbar.py

```diff
@@ -55,6 +54,13 @@
 _log = logging.getLogger(__name__)
 
 _make_axes_param_doc = """
+    location : None or {'left', 'right', 'top', 'bottom'}
+        The position, relative to the parent axes, where the colorbar axes is
+        created.  If None, the value will either come from the *orientation* if
+        it is set, or default to 'right'.
+    orientation : None or {'vertical', 'horizontal'}
```

Comment:
OK, this will be a little confusing, as I think we use `vertical` and `horizontal` all over the docs? (I haven't checked)

Also, its not clear from the doc what happens if the the user calls both?  Someone could conceivably think they can have a horizontal colorbar on the right.

Reply:
What's the thing that's confusing?  (Sorry, I really can't parse your comment...)
An error is thrown if both are set (regardless of whether they are consistent or not).  I changed that to only throw if the two values are inconsistent (e.g. right/horizontal), and silently accept consistent values.
I expanded a bit the docstring to document the behavior, it's a bit verbose though...

Reply:
You are saying it’s preferable to not use “vertical”, but I think we do that often in the docs.

Reply:
I already weakened a bit the language in the new version, if you have something better feel free to propose it.